application/src/main/java/org/opentripplanner/updater/siri/updater/lite/SiriETLiteHttpTripUpdateSource.java

```diff
@@ -0,0 +1,93 @@
+package org.opentripplanner.updater.siri.updater.lite;
+
+import static org.opentripplanner.updater.trip.UpdateIncrementality.FULL_DATASET;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Optional;
+import org.opentripplanner.framework.io.OtpHttpClientException;
+import org.opentripplanner.updater.siri.updater.EstimatedTimetableSource;
+import org.opentripplanner.updater.siri.updater.SiriFileLoader;
+import org.opentripplanner.updater.siri.updater.SiriLoader;
+import org.opentripplanner.updater.spi.HttpHeaders;
+import org.opentripplanner.updater.trip.UpdateIncrementality;
+import org.opentripplanner.utils.tostring.ToStringBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.org.siri.siri20.Siri;
+
+/**
+ * SIRI Lite downloads periodically all messages as a single GET request.
+ */
+public class SiriETLiteHttpTripUpdateSource implements EstimatedTimetableSource {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SiriETLiteHttpTripUpdateSource.class);
+  /**
+   * The framework code requires a requestor ref but in SIRI Lite this is not used.
+   */
+  private static final String DUMMY_REQUESTOR_REF = "OpenTripPlanner";
+
+  private final Parameters parameters;
+
+  private final SiriLoader siriLoader;
+
+  public SiriETLiteHttpTripUpdateSource(Parameters parameters) {
+    this.parameters = parameters;
+
+    this.siriLoader = createLoader(parameters);
+  }
+
+  @Override
+  public Optional<Siri> getUpdates() {
+    try {
+      var siri = siriLoader.fetchETFeed(DUMMY_REQUESTOR_REF);
+      if (siri.map(Siri::getServiceDelivery).isEmpty()) {
+        return Optional.empty();
+      }
+      return siri;
+    } catch (OtpHttpClientException e) {
+      LOG.warn("Could not get SIRI-ET data from {}", parameters.uri(), e);
+    } catch (Exception e) {
+      LOG.warn("Failed to parse SIRI-ET feed from {}", parameters.uri(), e);
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public UpdateIncrementality incrementalityOfLastUpdates() {
+    return FULL_DATASET;
+  }
+
+  public String toString() {
+    return ToStringBuilder
+      .of(this.getClass())
+      .addStr("feedId", parameters.feedId())
+      .addStr("uri", parameters.uri().toString())
+      .toString();
+  }
+
+  private static SiriLoader createLoader(Parameters parameters) {
```

Comment:
DRY - This can be refactored out. Either create a SiriLoaderFactory or create it in the UpdatorConfigurator and pass it in.

Reply:
I've done both in https://github.com/opentripplanner/OpenTripPlanner/pull/6284/commits/c7bf5ed639ff33f7e952567bcae2b8fd01565fd4